api/v1alpha1/dnspolicy_types.go

```diff
@@ -178,6 +180,21 @@ func (p *DNSPolicy) GetStatus() kuadrantgatewayapi.PolicyStatus {
 
 func (p *DNSPolicy) Kind() string { return p.TypeMeta.Kind }
 
+func (p *DNSPolicy) List(ctx context.Context, c client.Client, namespace string) []kuadrantgatewayapi.Policy {
+	policyList := &DNSPolicyList{}
+	listOptions := &client.ListOptions{Namespace: namespace}
+	err := c.List(ctx, policyList, listOptions)
+	if err != nil {
+		return []kuadrantgatewayapi.Policy{}
+	}
+	policies := make([]kuadrantgatewayapi.Policy, 0)
```

Comment:
Any thoughts an setting the capacity here and in the other `List()` functions since we know the policy length being returned? 
```suggestion
	policies := make([]kuadrantgatewayapi.Policy, 0, len(policyList))
```

Reply:
should be update now